### PR TITLE
Update Examples Browser NPM dependencies

### PR DESCRIPTION
**Pull request by willeastcott**
Update all NPM dependencies except `react-router-dom` (requires some work to upgrade).

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: examples/scripts/build-examples.mjs [hunk 1]
```diff
@@ -43,7 +43,8 @@ const generateExampleFile = (categoryKebab, exampleNameKebab, setEngineType, fil
     const engine = engineFor(engineType);
     html = html.replace(/'@ENGINE'/g, JSON.stringify(engine));
 
-    if (/'@([A-Z0-9_]+)'/g.test(html)) {
+    /* eslint-disable-next-line regexp/no-unused-capturing-group */
+    if (/'@([A-Z0-9_]+)'/.test(html)) {
```
Comment (kungfooman) [inline examples/scripts/build-examples.mjs:47]: Same here

File: examples/scripts/build-examples.mjs [hunk 2]
```diff
@@ -114,6 +115,8 @@ export const build = (env = {}) => {
 
             const scriptPath = resolve(path, `${exampleNameKebab}.${file}`);
             let script = fs.readFileSync(scriptPath, 'utf-8');
+
+            /* eslint-disable-next-line regexp/no-unused-capturing-group */
```
[the remark this answers is not in the document]
Comment (willeastcott) [reply examples/scripts/build-examples.mjs:119]: This is just to get the upgrade done. If someone wants to go in and resolve it afterwards, be my guest. But I didn't want to disable the rule globally.

File: examples/scripts/build-examples.mjs (hunk 1 above)
Comment (willeastcott) [reply examples/scripts/build-examples.mjs:47]: See me previous comment.

File: examples/scripts/build-examples.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (willeastcott) [reply examples/scripts/build-examples.mjs]: Actually, I've gone ahead and fixed the problem properly.

File: examples/scripts/build-examples.mjs (hunk 1 above)
Comment (willeastcott) [reply examples/scripts/build-examples.mjs]: I've fixed this one too.

File: examples/scripts/build-examples.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kungfooman) [reply examples/scripts/build-examples.mjs]: Niice, thank you :+1:

**Review by kpal81xd — APPROVED** [review]
Fixed regex parser issues

**Pull request merged**